### ISSUE TITLE
DataTable: add support for multiple dynamic columns

### ISSUE THREAD

**Issue by cmoetzing** [primefaces/primefaces#2154]
## 1) Environment
- Primefaces 6.1RC1
- Wildfly 10.1.Final
- All Browsers
- Mojarra 2.2.13

## 2) Expected behavior
When placing two `p:colunns` tags inside a `p:dataTable` nested inside another interator (`p:repeat`, `p:dataTable`) both dynamic sets of columns should be displayed and all iterated tables should display their own columns.

## 3) Actual behavior
The first table defines the structure of the table e.g. the amount of columns and column labels. All subsequent tables will display these columns, not teir own.

## 4) Steps to reproduce
See example.

## 5) Sample XHTML
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:p="http://primefaces.org/ui"
    xmlns:c="http://java.sun.com/jsp/jstl/core">

<h:head>
    <title>DataTable BUG</title>
</h:head>

<body>
    <h:form id="form">
        <p:dataTable id="families" value="#{backingBean.families}" var="family">
            <p:column headerText="Families">
                <h3>#{family.name}</h3>
                <p:dataTable id="persons" value="#{family.persons}" var="person">
                    <p:column headerText="Name">
                        Does #{person} ...
                    </p:column>
                    <p:columns value="#{family.cars}" var="car" headerText="#{car}">
                        ... have insurance for the #{car}? 
                    </p:columns>
                    <p:columns value="#{backingBean.countries}" var="country" headerText="#{country}">
                        ... have insurance for #{country}?
                    </p:columns>
                </p:dataTable>
            </p:column>    
       </p:dataTable>
    </h:form>   
</body>
</html>
```

## 6) Sample bean
```java
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.faces.view.ViewScoped;
import javax.inject.Named;

@Named
@ViewScoped
public class BackingBean implements Serializable {
    private static final long serialVersionUID = 1L;

    private List<Family> families;
    private List<String> countries;

    public void generateFamilies() {
        this.families = new ArrayList<Family>();

        Family f = new Family("Beta");
        f.getCars().add("Mercedes");
        f.getCars().add("Porsche");
        f.getCars().add("VW");
        f.getCars().add("Dodge");
        f.getPersons().add("he");
        f.getPersons().add("she");
        families.add(f);

        f = new Family("Alpha");
        f.getCars().add("BMW");
        f.getPersons().add("he");
        f.getPersons().add("she");
        families.add(f);
    }

    private void generateCountries() {
        this.countries = new ArrayList<String>();
        this.countries.add("Germany");
        this.countries.add("England");
        this.countries.add("Spain");
    }

    public List<Family> getFamilies() {
        if (null == families) {
            generateFamilies();
        }
        return families;
    }

    public List<String> getCountries() {
        if (null == countries) {
            generateCountries();
        }
        return countries;
    }

    public class Family {
        private String name;
        private List<String> persons;
        private List<String> cars;

        public Family(String name) {
            this.name = name;
            this.persons = new ArrayList<String>();
            this.cars = new ArrayList<String>();
        }

        public String getName() {
            return name;
        }

        public List<String> getPersons() {
            return persons;
        }

        public List<String> getCars() {
            return cars;
        }
    }

}
```

## 7) Sample images
[This post](http://stackoverflow.com/questions/42549648/nested-datatable-with-multiple-dynamic-columns-display-same-data-for-each-row) contains images of how the pages looks like when the example is executed.

**Comment by Rapster** [comment]
Are you sure this is what you wanna do? Nested table? It might cause performance issue, and the rendering might not be so fancy. I'm just working around your problem, but have you tried this http://www.primefaces.org/showcase/ui/data/datatable/subTable.xhtml

**Comment by cmoetzing** [comment]
Unfortunately that is what I <s>want</s> need to do. I know that there are workarounds. I could also wrap my 3 different objects in one wrapper, create one `p:columns` iterating over the wrapper and render that one column depending on the content of the wrapper. But that is ugly.

I think subtables do not solve my problem. I need x separate tables. And the columns of these tables vary. In the most extreme case they all share only one column with the same title. As I understand `p:subTable` I would still need two `p:columns` tags inside my table.

**Comment by kukel** [comment]
Tried 6.1RC1 ? There have been fixes in this area since 6.0 was release iirc

**Comment by cmoetzing** [comment]
6.1RC1 still behaves the same (updated issue aswell).

**Comment by tandraschko** [comment]
"new feature" as the idea was to have 1 p:columns

**Comment by melloware** [comment]
This has been implemented in 9.0 by @tandraschko  See: https://github.com/primefaces/primefaces/issues/6659
